rfc/rfc-0007-resource-references.md

```diff
@@ -0,0 +1,40 @@
+# RFC-0007: Resource References
+
+**Authors:** Scott Andrews
+
+**Status:**
+
+**Pull Request URL:** [#1370](https://github.com/projectriff/riff/pull/1370)
+
+**Superseded by:** N/A
+
+**Supersedes:** N/A
+
+**Related:** [RFC 0006: Guidance for child resource names](https://github.com/projectriff/riff/pull/1369)
+
+
+## Problem
+It's common for one resource to reference another. For example, a Deployer can reference a Function build to obtain the latest image, or a stream provider references the gateway Service it created. riff should have a common way to express relationships between Kubernetes resources.
+
+### Anti-Goals
+This RFC is not define how specific riff resources reference other resources, it is only providing guidance.
+
+## Solution
+The Kubernetes API project provides two mechanisms for referencing other resources. ObjectReference and LocalObjectReference.
+
+[ObjectReference](https://godoc.org/k8s.io/api/core/v1#ObjectReference) has full support for referencing any k8s resource, commonly by apiGroup, kind, namespace, and name.
+
+[LocalObjectReference](https://godoc.org/k8s.io/api/core/v1#LocalObjectReference) contains only a resource's name. The apiGroup, kind and namespace must be inferred.
```

Comment:
I haven't seen uses of `LocalObjectReference` - do you have examples?

Reply:
Note that the `TODO` in the `LocalObjectReference` struct may result in `LocalObjectReference` being deprecated in favour of `ObjectReference`.

Reply:
@dprotaso https://sourcegraph.com/github.com/kubernetes/api/-/blob/core/v1/types.go#L5024:6&tab=references